### PR TITLE
Implement Reply to Slack Thread for Answering Questions

### PR DESCRIPTION
**Pull request by TakeMeGH**
**Changes Made:**

- **Functionality Added:** 
  - Implemented the `reply_to_slack_thread` function, allowing replies to messages in Slack threads.
  - Utilized the `chat.postMessage` API method for sending replies.

- **Testing:**
  - Update unit tests to ensure the reliability of the `reply_to_slack_thread` function.
  - Verified the correctness of channel and thread ID handling in the tests.

### CODE REVIEW

**Comment by jasonirvine76** [comment]
Looks good to me for now

**Comment by jasonirvine76** [comment]
Is there any character limit on the response or the request?

**Comment by TakeMeGH** [comment]
> Is there any character limit on the response or the request?

For now, I think there are no limitations yet. Perhaps when integrating with the chatbot, limitations from the chatbot itself might appear.

File: adapter/slack.py [hunk 1]
```diff
@@ -28,11 +28,17 @@ async def ask(self, request: Request):
             _, question = parts
         else:
             raise HTTPException(status_code=400, detail="Missing parameter in the request.")
+        
+        question = f"{question} \n\n<@{user_id}>"
 
-            
-        user_info = self.app.client.users_info(user=user_id)
-        display_name = user_info["user"]["profile"]["display_name"]
+        response = self.app.client.chat_postMessage(channel=channel_id, text=question)
```
[the remark this answers is not in the document]
Comment (TakeMeGH) [reply adapter/slack.py:34]: Yeah, i think that line of code can fail. Good catch! I will include a try/catch block in that code.

[the remark this answers is not in the document]
Comment (TakeMeGH) [reply adapter/slack.py:34]: Yeah, as you and alex said that code will not always succeed. I will include a try/catch block in that code.

**Pull request merged**